pkg/translator/azure/resourcelogs_to_logs.go

```diff
@@ -207,6 +209,22 @@ func extractRawAttributes(log azureLogRecord) map[string]any {
 	return attrs
 }
 
+func copyProperties(category string, properties *any, attrs map[string]any) {
+	pmap := (*properties).(map[string]any)
+	var attrsProps map[string]any = nil
+	for k, v := range pmap {
+		if otelKey, found := ResourceLogKeyToSemConvKey(k, category); found {
+			attrs[otelKey] = v
+		} else {
+			if attrsProps == nil {
```

Comment:
Instead of doing this inside the loop we could instantiate the map before the loop, inside the else block add properties, and then when the loop is finished check if the length of the map is greater than 0 and add it to `azure.properties` if true.

Reply:
I was trying to avoid the allocation unless needed; is that an over-optimization?

Reply:
Ah. An empty map won't be allocated on the heap, right?